src/components/confirm/confirm.pw.tsx

```diff
@@ -0,0 +1,648 @@
+      );
+    });
+  });
+
+  testData.forEach((subTitleText) => {
+    test(`should check subtitle text is ${subTitleText}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<ConfirmComponent subtitle={subTitleText} />);
+
+      await expect(page.getByText(subTitleText)).toHaveAttribute(
+        "data-element",
+        "subtitle"
+      );
+    });
+  });
+
+  (heights as [number, string][]).forEach(([heightnumber, heightstring]) => {
+    test(`should check Confirm height is ${heightstring}px`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<ConfirmComponent height={heightstring} />);
+
+      const viewportHeight = 768;
+
+      let resultHeight: number;
+      if (heightnumber >= viewportHeight - 20) {
+        resultHeight = viewportHeight - 20;
+      } else {
+        resultHeight = heightnumber;
+      }
+
+      await assertCssValueIsApproximately(
+        page.getByRole("alertdialog"),
+        "height",
+        resultHeight
+      );
+    });
+  });
+
+  ([
+    [SIZE.EXTRASMALL, 300],
+    [SIZE.SMALL, 380],
+    [SIZE.MEDIUMSMALL, 540],
+    [SIZE.MEDIUM, 750],
+    [SIZE.MEDIUMLARGE, 850],
+    [SIZE.LARGE, 960],
+    [SIZE.EXTRALARGE, 1080],
+  ] as [ConfirmProps["size"], number][]).forEach(([sizeName, size]) => {
+    test(`should check confirm size is %{sizeName} with width of ${size}px`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<ConfirmComponent size={sizeName} />);
+
+      await assertCssValueIsApproximately(
+        page.getByRole("alertdialog"),
+        "width",
+        size
+      );
+    });
+  });
+
+  ([
+    ["error", "error"],
+    ["warning", "warning"],
+  ] as [ConfirmProps["iconType"], string][]).forEach(([iconType, element]) => {
+    test(`should check confirm has ${iconType} icon`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<ConfirmComponent iconType={iconType} />);
+
+      const firstIcon = icon(page).first();
+      await expect(firstIcon).toHaveAttribute("data-element", element);
+    });
+  });
+
+  confirmButtonTypes.forEach(([type, colour, border]) => {
+    test(`should check confirm button is ${type} type`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<ConfirmComponent confirmButtonType={type} />);
+
+      const button = page.getByRole("button").filter({ hasText: "Yes" });
+      await expect(button).toHaveCSS("color", colour);
+      await expect(button).toHaveCSS("border-color", border);
+    });
+  });
+
+  cancelButtonTypes.forEach(([type, colour, border]) => {
+    test(`should check cancel button is ${type} type`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<ConfirmComponent cancelButtonType={type} />);
+
+      const button = page.getByRole("button").filter({ hasText: "No" });
+      await expect(button).toHaveCSS("color", colour);
+      await expect(button).toHaveCSS("border-color", border);
+    });
+  });
+
+  ([
+    [true, "disabled"],
```

Comment:
suggestion: as this same array is used below as well maybe it would be a good approach to define it in a const like so

```
const disabledStates: [[ConfirmProps["disableCancel"] | [ConfirmProps["disableConfirm"], string][] = [
    [true, "disabled"],
    [false, "enabled"],
]